google/cloud/bigquery/magics/magics.py

```diff
@@ -720,6 +736,39 @@ def _cell_magic(line, query):
     finally:
         close_transports()
 
+    if args.send_widget_job:
```

Comment:
i'm confused why this async query block is located here. the code already constructs a client and job_config object and calls client.query in the _run_query() method. i'd expect that the send_widget_job boolean be passed as an additional argument there, and the thread code to be inside there.
also, this cell_magic function is to return the results dataframe. not sure how that happens in the threaded case.

Reply:
I'm running and tweaking in Jupyterlab!

Reply:
You're right - I removed the redundant client instance! As for the dataframe, I'm tweaking the output in Jupyterlab currently and will update with what the current solution looks like for further feedback!

Reply:
for handling the output dataframe, it may be sufficient to populate the `args.destination_var` variable in your thread func (if provided) instead of returning it (since the magic function would return while the job is still running, returning the result is not possible, i'd think).